src/rlx_resolve.erl

```diff
@@ -56,20 +56,32 @@ subset(Apps, World, LibDirs, CheckCodeLibDirs) ->
 subset([], _World, _Seen, _LibDirs, _CheckCodeLibDirs, Acc) ->
     Acc;
 subset([Goal | Rest], World, Seen, LibDirs, CheckCodeLibDirs, Acc) ->
-    {Name, Vsn} = name_version(Goal),
+    {Name, Vsn, Optional} = name_version(Goal),
     case sets:is_element(Name, Seen) of
         true ->
             subset(Rest, World, Seen, LibDirs, CheckCodeLibDirs, Acc);
         _ ->
-            AppInfo=#{applications := Applications,
-                      included_applications := IncludedApplications} =
-                find_app(Name, Vsn, World, LibDirs, CheckCodeLibDirs),
-            subset(Rest ++ Applications ++ IncludedApplications,
+            {Rest1, Acc1} =
+                case find_app(Name, Vsn, World, LibDirs, CheckCodeLibDirs) of
+                    not_found when Optional == true ->
```

Comment:
```suggestion
                    not_found when Optional ->
```

Note that this omits the little weird case where an `included_application` can't logically be an `optional_application` but this validation allows an included app to miss if optional. I don't think we should really bother fixing that aspect yet.